org_fedora_oscap/common.py

```diff
@@ -341,6 +342,47 @@ def run_oscap_remediate(profile, fpath, ds_id="", xccdf_id="", tailoring="",
     return proc.stdout
 
 
+def _schedule_firstboot_remediation(
+        chroot, profile, ds_path, results_path, report_path, ds_id, xccdf_id, tailoring_path):
+    config = textwrap.dedent(f"""\
+    OSCAP_REMEDIATE_DS='{ds_path}'
+    OSCAP_REMEDIATE_PROFILE_ID='{profile}'
+    OSCAP_REMEDIATE_ARF_RESULT='{results_path}'
+    OSCAP_REMEDIATE_HTML_REPORT='{report_path}'
+    OSCAP_REMEDIATE_VERBOSE_LOGS='/var/tmp/oscap_verbose.log'
+    """)
+    if ds_id:
+        config += "OSCAP_REMEDIATE_DATASTREAM_ID='{ds_id}'\n"
+    if xccdf_id:
+        config += "OSCAP_REMEDIATE_XCCDF_ID='{xccdf_id}'\n"
+    if tailoring_path:
+        config += "OSCAP_REMEDIATE_TAILORING='{tailoring_path}'\n"
+
```

Comment:
> @matejak
> 
> I struggle testing it.
> 
> Please add an example kickstart file on which this can be tested.

I found out that RHEL8 (#190) is a better system for testing, as there is a compatible OpenSCAP RPM publicly available.
So you don't need anything special, any kickstart that e.g. sets up the ANSSI minimal profile is good enough.
You just need to add a line that adds the repo with the fresh OpenSCAP, e.g.

```
repo --name oscap-copr --baseurl https://download.copr.fedorainfracloud.org/results/packit/OpenSCAP-openscap-1838/epel-8-x86_64/
```

You can switch the firstboot remediation off using ks `remediate = ...` syntax, but no reason for doing that, and it is on by default.

> 
> Which RHEL9 profile do you recommend to avoid partitioning?

I have used ANSSI Minimal

> 
> How did you manage to insert an updated version of OpenSCAP that contains the new OpenSCAP feature "first boot remediation" to the installer environment? I built an RPM with that and added in to update image, however, the installation failed due to no space left on device. Have you done any hack?

That happens automatically if that OpenSCAP is the newest one - you just need to add the respective repository.

Reply:
@matejak Thank you very much for reply. It helped a bit. The advice to test #190 instead and do it on RHEL 8 helped me, because I get `OSCAP addon: Scheduling firstboot remediation` message in `/tmp/anaconda.log`, which I didn't see there when I tried #189 on a RHEL9. On both situations, I can see the `/etc/system-update` and the oscap firstboot config file in `/mnt/sysroot/var/tmp`, so it seems that even on RHEL 9 something happens. However, I still struggle, because after the installation finishes and I click on the reboot button on Anaconda, the VM starts the installation again from the beginning. What do you have in your kickstart? Perhaps there can be something wrong in the kickstart that I have?

Reply:
I found out that I wasn't able to test this with reanaconda for some reason. I had to set up a brand new VM (I did that using virt-manager) that is able to properly boot the installed system after the installation is complete.
The addon does everything to trigger the firstboot event, but the target system needs the new oscap scanner installed to pick that event up.

Reply:
@matejak OK, thanks.

I was able to set up a RHEL 8.5 VM successfully, using this: 

```
virt-install --connect qemu:///system --name firstboot --memory 2048 --vcpus 2 --disk size=8 --os-variant rhel8.5  --location http://XXXXX/released/RHEL-8/8.5.0/BaseOS/x86_64/os/ -x inst.updates=http://192.168.122.1:8000/update.img  --initrd-inject ~/work/git/oscap-anaconda-addon/testing_files/firstboot-rhel8.cfg --extra-args="ks=file:/firstboot-rhel8.cfg" --network default
```
where `firstboot-rhel8.cfg` is:

```

# values saving a lot of clicks in the GUI
lang en_US.UTF-9
keyboard --xlayouts=us --vckeymap=us
timezone Europe/Prague
rootpw aaaaa
bootloader --location=mbr
clearpart --initlabel --all
autopart --type=plain
graphical

url --url=http://XXXX/released/RHEL-8/8.5.0/BaseOS/x86_64/os/
repo --name appstream --baseurl=http://XXXX/released/RHEL-8/8.5.0/AppStream/x86_64/os/
repo --name oscap --baseurl=https://download.copr.fedorainfracloud.org/results/packit/OpenSCAP-openscap-1838/epel-8-x86_64/

%packages
vim
%end

%addon org_fedora_oscap
    content-type = scap-security-guide
    profile = xccdf_org.ssgproject.content_profile_anssi_bp28_minimal
%end
```


Next time, because OAA PRs are complex and testing them require a specific set up of VMs, it would be amazing if OAA PRs contained instructions on how to test them to facilitate the reviewer's work. For example, used ks and/or command. I assume that it wouldn't be a problem for the PR authors because they have them ready at hand because they were testing the PR themselves on their machine right before they submitted the PR so they can only copy-paste it.

Reply:
What about adding a kickstart like this into the git repository?

Reply:
Isn't #191 good enough?

Reply:
Yes, It is, It's great! I'm sorry, I forgot that you added it in #191.